docker/docker_client.go

```diff
@@ -224,6 +225,7 @@ func newDockerClientFromRef(sys *types.SystemContext, ref dockerReference, write
 	}
 	client.username = username
 	client.password = password
+	client.registryToken = sys.DockerRegistryToken
```

Comment:
`*types.SystemContext` can always be `nil`, and must be treated the same as `&types.SystemContext{}`.

Reply:
fixed b6597e0